e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js

```diff
@@ -1482,3 +1482,41 @@ describe("issue 47887", () => {
     });
   });
 });
+
+describe("Issue 48851", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  const manyValues = Array(75)
+    .fill(0)
+    .map(() => Math.round(Math.random() * 1000_000_000).toString(36))
+    .join(", ");
+
+  function openTableHeaderMenu(name) {
+    cy.findAllByTestId("header-cell")
+      .contains(name)
+      .trigger("mousedown")
+      .trigger("mouseup");
+  }
+
+  it("should not overflow the filter popover, even when there are a lot of values (metabase#48851)", () => {
+    openProductsTable();
+    openTableHeaderMenu("Title");
+
+    popover().within(() => {
+      cy.findByText("Filter by this column").click();
+      cy.findByText("Is").click();
+    });
+
+    popover().last().findByText("Contains").click();
+    popover()
+      .first()
+      .findByPlaceholderText("Enter some text")
+      .type(manyValues, { timeout: 0 });
```

Comment:
This is a bit time-consuming.
Instead we could programmatically create a question with such filter applied, and later in UI we'd only click that filter to edit it - that should reproduce the bug much faster.

Reply:
This bug only shows up in the filter popover for _new_ filters on the header popover.

Clicking the existing filters already works as expected (albeit it _could_ look better, see comment below).

Clicking the header on a question that has this filter on it already will not show the existing filter, but it will create a new filter. Therefore we need to create the values manually, unfortunately.